handler/EventHandler.go

```diff
@@ -0,0 +1,225 @@
+package handler
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/ONSdigital/go-ns/log"
+	"github.com/coocood/freecache"
+)
+
+//struct for eventhandler which handles the instance and the start of the api
+type EventReport struct {
+	InstanceID string `avro:"instance_id"`
+	EventType  string `avro:"event_type"`
+	EventMsg   string `avro:"event_message"`
+}
+
+type config struct {
+	ImportAPIURL string
+	AuthToken    string
+}
+
+type instance struct {
+	InstanceID                string   `json:"instance_id"`
+	NumberOfObservations      int64    `json:"total_observations"`
+	TotalInsertedObservations int64    `json:"total_inserted_observations,omitempty"`
+	State                     string   `json:"state"`
+	Events                    *[]event `json:"events, omitempty"`
+}
+
+type event struct {
+	Type          string `json:"type"`
+	Message       string `json:"message"`
+	MessageOffset string `json:"messageOffset"`
+}
+
+var failed = "failed"
+
+//method to set an event not really needed but testing made things easier
+func (e *EventReport) EventSetter(InstanceID string, EventType string, EventMessage string) {
+	e.InstanceID = InstanceID
+	e.EventType = EventType
+	e.EventMsg = EventMessage
+}
+
+//main function that triggers everything else
+func (e *EventReport) HandleEvent(httpClient *http.Client, authToken string, c *freecache.Cache) error {
+	cfg := config{
+		ImportAPIURL: "http://localhost:21800",
+		AuthToken:    authToken,
+	}
+
+	status, events, err := e.checkInstance(httpClient, cfg)
+	if err != nil {
```

Comment:
add an error log message so we can see from the log what when wrong / where it went wrong.